test/unit/int_algolia/scripts/algolia/helper/AlgoliaJobReport.test.js

```diff
@@ -0,0 +1,117 @@
+const AlgoliaJobReport = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/AlgoliaJobReport');
+
+describe('AlgoliaJobReport', () => {
+    let jobReport;
+
+    beforeEach(() => {
+        jobReport = new AlgoliaJobReport('job-id', 'product');
+    });
+
+    describe('writeToCustomObject', () => {
+        it('should write the job report to a custom object', () => {
+            const CustomObjectMgr = require('dw/object/CustomObjectMgr');
+            const Transaction = require('dw/system/Transaction');
+            const System = require('dw/system/System');
+            const StringUtils = require('dw/util/StringUtils');
+
+            const customObjectID = 'job-id__' + StringUtils.formatCalendar(System.getCalendar(), 'yyMMdd-HHmmss');
+
+            CustomObjectMgr.createCustomObject.mockImplementation((objectType, id) => {
+                return {
+                    custom: {
+                        jobID: 'job-id',
+                        jobType: 'product',
+                        startTime: new Date(),
+                        endTime: new Date(),
+                        processedItems: 10,
+                        processedItemsToSend: 5,
+                        siteLocales: 2,
+                        recordsToSend: 20,
+                        recordsSent: 15,
+                        recordsFailed: 5,
+                        chunksSent: 3,
+                        chunksFailed: 1,
+                        error: false,
+                        errorMessage: '',
+                    },
+                };
+                // throw new Error('Invalid arguments');
```

Comment:
```suggestion
```

But you actually should remove this whole mockImplementation, as it doesn't represent what the actual `createCustomObject` does, and you already have a mock declared in jest.setup now